packages/kbn-config-schema/src/types/object_type.ts

```diff
@@ -30,17 +30,35 @@ export type TypeOf<RT extends Type<any>> = RT['type'];
 // this might not have perfect _rendering_ output, but it will be typed.
 export type ObjectResultType<P extends Props> = Readonly<{ [K in keyof P]: TypeOf<P[K]> }>;
 
+interface AllowUnknowns {
+  allowUnknowns: true;
+  ignoreUnknowns?: false;
+}
+
+interface IgnoreUnknowns {
+  allowUnknowns?: false;
+  ignoreUnknowns: true;
+}
+
+interface ForbidUnknowns {
+  allowUnknowns?: false;
+  ignoreUnknowns?: false;
+}
+// type check to not permit both to be set to `true` at the same time
+type UnknownOptions = AllowUnknowns | IgnoreUnknowns | ForbidUnknowns;
```

Comment:
I did this because it was less invasive, but it feels messy to me.

Since these options are mutually exclusive of each other, I think a const might be better:

```ts
interface UnknownOptions {
   unknowns?: 'allow' | 'ignore' | 'forbid'
}
```

with the default behavior being `forbid`.

I'm happy to make this change, but I wanted to discuss before doing so, as it'll touch a lot of downstream code.

Reply:
It makes sense to me...even it will require to touch 70 places in the codebase 🙈 welcome to the platform team world.